Build failure: python311Packages.nptyping

### Steps To Reproduce

Steps to reproduce the behavior:

1. Run `nix-shell -p python311Packages.nptyping`.
2. The build fails

### Build log

```
...

==================================== ERRORS ====================================
___________________ ERROR collecting tests/test_beartype.py ____________________
tests/test_beartype.py:13: in <module>
    @beartype
/nix/store/5w99h9myq282z67gzkvc6vc7ac4n1zkz-python3.11-beartype-0.17.2/lib/python3.11/site-packages/beartype/_decor/decorcache.py:77:>
    return beartype_object(obj, conf)
/nix/store/5w99h9myq282z67gzkvc6vc7ac4n1zkz-python3.11-beartype-0.17.2/lib/python3.11/site-packages/beartype/_decor/decorcore.py:83: >
    _beartype_object_fatal(obj, conf=conf, **kwargs)
/nix/store/5w99h9myq282z67gzkvc6vc7ac4n1zkz-python3.11-beartype-0.17.2/lib/python3.11/site-packages/beartype/_decor/decorcore.py:132:>
    beartype_nontype(obj, **kwargs)  # type: ignore[return-value]
/nix/store/5w99h9myq282z67gzkvc6vc7ac4n1zkz-python3.11-beartype-0.17.2/lib/python3.11/site-packages/beartype/_decor/_decornontype.py:>
    return beartype_func(obj, **kwargs)  # type: ignore[return-value]
/nix/store/5w99h9myq282z67gzkvc6vc7ac4n1zkz-python3.11-beartype-0.17.2/lib/python3.11/site-packages/beartype/_decor/_decornontype.py:>
    func_wrapper_code = generate_code(bear_call)
/nix/store/5w99h9myq282z67gzkvc6vc7ac4n1zkz-python3.11-beartype-0.17.2/lib/python3.11/site-packages/beartype/_decor/wrap/wrapmain.py:>
    code_check_params = _code_check_args(bear_call)
/nix/store/5w99h9myq282z67gzkvc6vc7ac4n1zkz-python3.11-beartype-0.17.2/lib/python3.11/site-packages/beartype/_decor/wrap/wrapmain.py:>
    reraise_exception_placeholder(
/nix/store/5w99h9myq282z67gzkvc6vc7ac4n1zkz-python3.11-beartype-0.17.2/lib/python3.11/site-packages/beartype/_util/error/utilerrorrai>
    raise exception.with_traceback(exception.__traceback__)
/nix/store/5w99h9myq282z67gzkvc6vc7ac4n1zkz-python3.11-beartype-0.17.2/lib/python3.11/site-packages/beartype/_decor/wrap/wrapmain.py:>
    hint_sane = sanify_hint_root_func(
/nix/store/5w99h9myq282z67gzkvc6vc7ac4n1zkz-python3.11-beartype-0.17.2/lib/python3.11/site-packages/beartype/_check/convert/convsanif>
    hint = reduce_hint(
/nix/store/5w99h9myq282z67gzkvc6vc7ac4n1zkz-python3.11-beartype-0.17.2/lib/python3.11/site-packages/beartype/_check/convert/convreduc>
    hint = _reduce_hint_uncached(
/nix/store/5w99h9myq282z67gzkvc6vc7ac4n1zkz-python3.11-beartype-0.17.2/lib/python3.11/site-packages/beartype/_check/convert/convreduc>
    hint_sign = get_hint_pep_sign_or_none(hint)
/nix/store/5w99h9myq282z67gzkvc6vc7ac4n1zkz-python3.11-beartype-0.17.2/lib/python3.11/site-packages/beartype/_util/cache/utilcachecal>
    raise exception
/nix/store/5w99h9myq282z67gzkvc6vc7ac4n1zkz-python3.11-beartype-0.17.2/lib/python3.11/site-packages/beartype/_util/cache/utilcachecal>
    return_value = args_flat_to_return_value[args_flat] = func(
/nix/store/5w99h9myq282z67gzkvc6vc7ac4n1zkz-python3.11-beartype-0.17.2/lib/python3.11/site-packages/beartype/_util/hint/pep/utilpepge>
    die_if_hint_pep604_inconsistent(hint)
/nix/store/5w99h9myq282z67gzkvc6vc7ac4n1zkz-python3.11-beartype-0.17.2/lib/python3.11/site-packages/beartype/_util/hint/pep/proposal/>
    raise BeartypeDecorHintPep604Exception(
E   beartype.roar.BeartypeDecorHintPep604Exception: Type hint NDArray[Shape['2, 2'], Float] inconsistent with respect to repr() strin>
E       >>> repr(NDArray[Shape['2, 2'], Float])
E       NDArray[Shape['2, 2'], Float]  # <-- this is fine
E       >>> repr(NDArray[Shape['2, 2'], Float] | int)
E       nptyping.ndarray.NDArray | int  # <-- *THIS IS REALLY SUPER BAD*
E   
E       # Ideally, that output should instead resemble:
E       >>> repr(NDArray[Shape['2, 2'], Float] | int)
E       NDArray[Shape['2, 2'], Float] | int  # <-- what @beartype wants!

...
```

### Additional context

*N.A.*

### Notify maintainers

@bcdarwin

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[gavin@gavin-laptop-nixos-1:~/repos/nixpkgs]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.23, NixOS, 24.05 (Uakari), 24.05.20240403.08b9151`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/nix/store/wx38bcq0bi8xhia8yq21syxlkf65fk4j-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/nptyping/default.nix
+++ b/pkgs/development/python-modules/nptyping/default.nix
@@ -51,6 +51,8 @@ buildPythonPackage rec {
     "tests/test_typeguard.py"
     # tries to build wheel of package, broken/unnecessary under Nix:
     "tests/test_wheel.py"
+    # beartype fails a type check
+    "tests/test_beartype.py"
   ];
 
   pythonImportsCheck = [

